src/lifecycle.js

```diff
@@ -12,7 +12,12 @@ function sandbox (data, deserialize) {
       }
     }
 
-    action.resolve(payload)
+    // Strip out keys not managed by this repo. This prevents
+    // children from accidentally having their keys reset by
+    // parents.
+    let sanitary = repo.realm.prune({}, payload)
+
+    action.resolve(sanitary)
```

Comment:
Prune loops over all of the domains, copying over their keys from one object to another. This gets rid of any keys that aren't within what are specified when you call `repo.addDomain`.

The first argument, `{}` here, is the object to copy the values into. It's sort of like a base state.